src/plugins/visualizations/public/vis_types/vis_type_alias_registry.ts

```diff
@@ -65,19 +68,38 @@ export interface VisTypeAlias {
   };
 }
 
-const registry: VisTypeAlias[] = [];
+let registry: VisTypeAlias[] = [];
+let discardOnRegister: string[] = [];
 
 interface VisTypeAliasRegistry {
   get: () => VisTypeAlias[];
   add: (newVisTypeAlias: VisTypeAlias) => void;
+  remove: (visTypeAliasName: string) => void;
 }
 
 export const visTypeAliasRegistry: VisTypeAliasRegistry = {
   get: () => [...registry],
   add: (newVisTypeAlias) => {
-    if (registry.find((visTypeAlias) => visTypeAlias.name === newVisTypeAlias.name)) {
-      throw new Error(`${newVisTypeAlias.name} already registered`);
+    // if it exists on discardOnRegister array then we don't allow it to be registered
+    if (discardOnRegister.some((aliasName) => aliasName === newVisTypeAlias.name)) {
+      discardOnRegister = discardOnRegister.filter(
```

Comment:
Good catch @flash1293, I actually want to remove this. Thanx for this 💯

Reply:
What's the reasoning for removing it in the first place? I can't see a good use case for keeping this conditional branch. If `remove` is called once for a name, it would make sense to block all subsequent tries to add it as well, right?

Reply:
Yes this is how I had done it on the beginning but Tim suggested to change it https://github.com/elastic/kibana/pull/79627#discussion_r509139643 to prevent any potential race condition

Reply:
oh you mean that there is no reason to remove it, I see

Reply:
Done! 😄